### PR TITLE
Reenabling enter button on login type page new sign up

### PR DESCRIPTION
**Pull request by hannahbergam**
I mistakenly believed that updating the button to be of type 'submit' in [this pr](https://github.com/code-dot-org/code-dot-org/pull/62202) would mean that the keyboard functionality would be maintained. However, I've since realized that the submit type only works if the focus state is already on the button. 

This pr is largely a revert of that one, while keeping the proper submit type. I was also able to refactor the approach a bit now that Denys has enabled onKeyDown for inputs. Thanks @drizco for the tip!

Here is a video to show that the accessible experience still works (after the first click into the email field so I didn't have to show the whole header tabbing, I tabbed between fields and pressed enter to submit). 

https://github.com/user-attachments/assets/41d2b3cf-df1a-42df-8f6c-0c2717866c55

## Links

- spec: []()
- jira ticket: https://codedotorg.atlassian.net/browse/ACQ-2700

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/signUpFlow/LoginTypeSelection.tsx [hunk 1]
```diff
@@ -70,6 +70,24 @@ const LoginTypeSelection: React.FunctionComponent = () => {
...
+          button.click();
+        }
+      }
+    };
+
+    document.addEventListener('keydown', handleKeyDown);
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  }, []);
```
Comment (drizco) [inline apps/src/signUpFlow/LoginTypeSelection.tsx]: Love the event listener cleanup here! 

I think there's a more react-y way to do this now. the `TextField` component recently added support for an `onKeyDown` handler. and I think you can call `submitLoginType` directly without having to get the button by id

Comment (hannahbergam) [reply apps/src/signUpFlow/LoginTypeSelection.tsx]: Would an onKeyDown require the focus state to be on that input though? I believe the intention was to allow for an 'enter' to submit if the button is enabled from anywhere on the page

Comment (hannahbergam) [reply apps/src/signUpFlow/LoginTypeSelection.tsx]: Ahh or I would just add it to all three inputs?

Comment (drizco) [reply apps/src/signUpFlow/LoginTypeSelection.tsx]: oh interesting! yeah my thought was to add it to each of the inputs. I'm trying to think of a scenario where the user completes the form and then moves focus away from the buttons and inputs, and then presses enter 🤔 

It's certainly possible, but I'm wondering how necessary it is from an accessibility point of view?

Comment (drizco) [reply apps/src/signUpFlow/LoginTypeSelection.tsx]: in fact, what happens if a user using a screen reader completes the form but then tabs to a link or another button and hits enter?

Comment (hannahbergam) [reply apps/src/signUpFlow/LoginTypeSelection.tsx]: I agree that the three inputs specifically is probably the right move (and have it working, though I cannot for the life of me get this test to work with a focus state).

Comment (hannahbergam) [reply apps/src/signUpFlow/LoginTypeSelection.tsx]: Fixed!

**Review by drizco — APPROVED** [review]
🚀

**Pull request merged**